### PR TITLE
Build the frontend using webpack

### PR DESCRIPTION
**Pull request by benmccann**
This is a much more realistic example of how an app would be structured because it loads 1 JS file in production instead of 6.

### CODE REVIEW

File: front-end/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const webpack = require('webpack');
+const path = require('path');
+
+const APP_DIR = path.resolve(__dirname, 'src/main/resources/assets/');
+const BUILD_DIR = process.env.BUILD_SYSTEM.toLowerCase() === 'maven'
+    ? path.resolve(__dirname, 'target/classes/public/')
+    : path.resolve(__dirname, 'target/web/public/main/');
+
```
Comment (TimMoore) [inline front-end/webpack.config.js]: I'm getting an error running with SBT:

```
[error] /Users/tmoore/Projects/lagom/activator-lagom-java-chirper/front-end/webpack.config.js:5
[error] const BUILD_DIR = process.env.BUILD_SYSTEM.toLowerCase() === 'maven'
[error]                                           ^
[error] 
[error] TypeError: Cannot read property 'toLowerCase' of undefined
[error]     at Object.<anonymous> (/Users/tmoore/Projects/lagom/activator-lagom-java-chirper/front-end/webpack.config.js:5:43)
[error]     at Module._compile (module.js:569:30)
[error]     at Object.Module._extensions..js (module.js:580:10)
[error]     at Module.load (module.js:503:32)
[error]     at tryModuleLoad (module.js:466:12)
[error]     at Function.Module._load (module.js:458:3)
[error]     at Module.require (module.js:513:17)
[error]     at require (internal/module.js:11:18)
[error]     at /Users/tmoore/Projects/lagom/activator-lagom-java-chirper/front-end/target/webpack/contexts.js:4:19
[error]     at Object.<anonymous> (/Users/tmoore/Projects/lagom/activator-lagom-java-chirper/front-end/target/webpack/contexts.js:18:3)
com.github.stonexx.sbt.webpack.SbtWebpack$NodeExecuteFailureException: Failed to execute node.
	at com.github.stonexx.sbt.webpack.SbtWebpack$.com$github$stonexx$sbt$webpack$SbtWebpack$$runNode(SbtWebpack.scala:247)
	at com.github.stonexx.sbt.webpack.SbtWebpack$$anonfun$getContexts$1.apply(SbtWebpack.scala:141)
	at com.github.stonexx.sbt.webpack.SbtWebpack$$anonfun$getContexts$1.apply(SbtWebpack.scala:135)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (front-end/*:webpack) com.github.stonexx.sbt.webpack.SbtWebpack$NodeExecuteFailureException: Failed to execute node.
```

**Review by TimMoore — CHANGES_REQUESTED** [review]
I'm not very familiar with webpack, but this seems like a nice improvement to the front-end build.

Sorry that it took so long to review. It looks good other than the SBT issue (which I suppose is just a matter of configuring the [`envVars`](https://github.com/stonexx/sbt-webpack/blob/master/src/main/scala/com/github/stonexx/sbt/webpack/SbtWebpack.scala#L36) setting).

File: front-end/webpack.config.js (hunk 1 above)
Comment (benmccann) [reply front-end/webpack.config.js]: Fixed. Thanks for testing it out!

**Comment by TimMoore** [comment]
Just saw your change, @benmccann ... looks good. I was working on a local fix as well:

```diff
diff --git i/build.sbt w/build.sbt
index 76b012b..c4f7d15 100644
--- i/build.sbt
+++ w/build.sbt
@@ -84,6 +84,8 @@ lazy val frontEnd = project("front-end")
       (sourceDirectory in Assets).value :+
       (resourceDirectory in Assets).value,
 
+    WebpackKeys.envVars in webpack += "BUILD_SYSTEM" -> "sbt",
+
     // Remove to use Scala IDE
     EclipseKeys.createSrc := EclipseCreateSrc.ValueSet(EclipseCreateSrc.ManagedClasses, EclipseCreateSrc.ManagedResources)
   )
```

What do you think? It might be good to be explicit.

I also noticed that there's a v1.2.0 of the sbt plugin that was released since you originally opened this. Do you think it's a good idea to update?

**Review by TimMoore — APPROVED** [review]

**Comment by TimMoore** [comment]
I've tested locally with both of these changes and it all seems to check out:

```diff
diff --git i/build.sbt w/build.sbt
index 76b012b..c4f7d15 100644
--- i/build.sbt
+++ w/build.sbt
@@ -84,6 +84,8 @@ lazy val frontEnd = project("front-end")
       (sourceDirectory in Assets).value :+
       (resourceDirectory in Assets).value,
 
+    WebpackKeys.envVars in webpack += "BUILD_SYSTEM" -> "sbt",
+
     // Remove to use Scala IDE
     EclipseKeys.createSrc := EclipseCreateSrc.ValueSet(EclipseCreateSrc.ManagedClasses, EclipseCreateSrc.ManagedResources)
   )
diff --git i/project/plugins.sbt w/project/plugins.sbt
index e2ed674..f9fdd06 100644
--- i/project/plugins.sbt
+++ w/project/plugins.sbt
@@ -1,4 +1,4 @@
 addSbtPlugin("com.lightbend.lagom" % "lagom-sbt-plugin" % "1.3.5")
-addSbtPlugin("com.github.stonexx.sbt" % "sbt-webpack" % "1.1.0")
+addSbtPlugin("com.github.stonexx.sbt" % "sbt-webpack" % "1.2.0")
 addSbtPlugin("com.lightbend.conductr" % "sbt-conductr" % "2.3.0")
 addSbtPlugin("com.typesafe.sbteclipse" % "sbteclipse-plugin" % "5.1.0")
```

For the record, I tested both sbt and Maven, and tested reload-on-change in both with `circuitbreaker.jsx`. Everything works great.

**Comment by benmccann** [comment]
Those changes seem fine to me!

**Comment by TimMoore** [comment]
I'm not sure what caused the latest build to fail. I'll try re-running it.

**Comment by benmccann** [comment]
Looks like it's passing now

**Review by ignasi35 — APPROVED** [review]

**Comment by ignasi35** [comment]
@fsat @edwardcallahan I'm good with merging this unless you think there's any blocker.

**Comment by edwardcallahan** [comment]
no concerns here, thanks @ignasi35

**Comment by ignasi35** [comment]
Thanks @benmccann for your patience.

**Pull request merged**